lib/apiservers/engine/backends/system.go

```diff
@@ -288,6 +288,10 @@ func (s *System) AuthenticateToRegistry(ctx context.Context, authConfig *types.A
 		registryAddress = registryAddress + "/v2/"
 	}
 
+	if !strings.HasPrefix(registryAddress, "https://") && !strings.HasPrefix(registryAddress, "http://") {
+		registryAddress = "https://" + registryAddress
```

Comment:
If we really want to support any and all protocols, we should capture it in an issue around writing our own custom url parser, which is definitely not within the scope of what we are trying to do right now.